src/python/pants/jvm/jdk_rules.py

```diff
@@ -114,17 +151,18 @@ async def setup_jdk(coursier: Coursier, jvm: JvmSubsystem, bash: BashBinary) ->
 
     if java_version_result.exit_code != 0:
         raise ValueError(
-            f"Failed to locate Java for JDK `{jvm.jdk}`:\n"
+            f"Failed to locate Java for JDK `{jdk}`:\n"
             f"{java_version_result.stderr.decode('utf-8')}"
         )
 
     java_version = java_version_result.stderr.decode("utf-8").strip()
     jre_major_version = parse_jre_major_version(java_version)
     if not jre_major_version:
         raise ValueError(
-            f"Pants was unable to parse the output of `java -version` for JDK `{jvm.jdk}`. "
-            "Please open an issue at https://github.com/pantsbuild/pants/issues/new/choose "
-            f"with the following output:\n\n{java_version}"
+            "Pants was unable to parse the output of `java -version` for JDK "
+            f"`{jdk.major_version}`. Please open an issue at "
```

Comment:
The primary reason why this code exists is to actually capture the version reported by a `system` interpreter: as it stands, the `JdkSetup` class will have an empty string for the major version for "system", which would break this error message.

Given that, I'm not sure that it's worth it to split the fields of `JdkSetup` into major version and `is_system`... it seems like it could continue to be a string, as it is today...? Since we parse the major version from the output, compatibility checks between two versions could make `JdkRequest`s and then use `JdkSetup.jre_major_version` to check compatibility. A downside though, is that all of the JVMs in a particular repository will be forced to be installed... I'm not sure that that is worth worrying about yet though.

Reply:
@stuhood Understood. I've reverted to having just a `version` parameter, and we can work on how we specify JDK versions in a follow-up.